pytorch_lightning/core/lightning.py

```diff
@@ -1966,6 +1965,23 @@ def model_size(self) -> float:
             )
         return get_model_size_mb(self)
 
+    @property
+    def use_amp(self) -> bool:
+        """Returns true if using AMP.
+
+        .. deprecated:: v1.6     This property was deprecated in v1.6 and will be removed in v1.8.
+        """
+        rank_zero_deprecation(
+            "`LightningModule.use_amp` was deprecated in v1.6 and will be removed in v1.8."
+            " Please use `Trainer.amp_backend`.",
+            stacklevel=5,
+        )
+        return self._use_amp
+
+    @use_amp.setter
+    def use_amp(self, use_amp: bool) -> None:
+        self._use_amp = use_amp
```

Comment:
add a deprecation warning here too.

the trainer can then set the attribute using `_use_amp` to avoid the deprecation warning being logged.

Reply:
> the trainer can then set the attribute using `_use_amp` to avoid the deprecation warning being logged.

Actually we don't even need this setter then

Reply:
theoretically users could set `use_amp` as its a regular attribute now, so the setter + deprecation warning is needed for completeness.